pkg/server/image_pull.go

```diff
@@ -473,6 +476,8 @@ const (
 func appendInfoHandlerWrapper(ref string) func(f containerdimages.Handler) containerdimages.Handler {
 	return func(f containerdimages.Handler) containerdimages.Handler {
 		return containerdimages.HandlerFunc(func(ctx context.Context, desc imagespec.Descriptor) ([]imagespec.Descriptor, error) {
+			desc.Annotations[targetImageDigestLabel] = desc.Digest.String()
```

Comment:
I may have missed it but is there any other handler I can annotate the image digest?

As for why, a Snapshotter implementation might need Image digest alongwith layer digests. As this handler was specially created for that purpose, I added the annotation here.

What are your thoughts?

Reply:
kk, no this is your annotation handler.. Let's update the description above for the handler to include digests for all image content media types, such as manifests and their child layers.

Reply:
@mikebrow right now it appends annotations only for `imagespec.MediaTypeImageManifest, containerdimages.MediaTypeDockerSchema2Manifest`. Are you saying I update it to include digests for all media types?

Reply:
> @mikebrow right now it appends annotations only for `imagespec.MediaTypeImageManifest, containerdimages.MediaTypeDockerSchema2Manifest`. Are you saying I update it to include digests for all media types?

No. Just asking / pointing out there are a number of new manifest types that will now be decorated.  oci image manifest, multi-arch, index, docker1, docker2...
 
> I also updated the description. Does that look ok?

yes, that description works.

Reply:
> No. Just asking / pointing out there are a number of new manifest types that will now be decorated. Or do the other types get filtered before this call..?

I see what you mean now! Good point. Let's only decorate these two manifests that stargz supports. I moved the decoration. Please check again and let me know if it LGTY.

Reply:
it does look better... 

now for the debug problem.. why is our integration test bugging out?

Reply:
It says its unable to connect to `containerd`

connection error: desc = "transport: Error while dialing dial unix /run/containerd-test/containerd.sock: connect: connection refused"

Reply:
right... and how would that happen based on the new line of code..

Reply:
```
mike@mike-VirtualBox:~/go/src/github.com/containerd/cri$ cat /tmp/test-integration/containerd.log 
tail:

time="2020-09-30T12:36:33.372551654-05:00" level=debug msg="fetch response received" digest="sha256:2ca5e69e244d2da7368f7088ea3ad0653c3ce7aaccd0b8823d11b0d5de956002" mediatype=application/vnd.docker.distribution.manifest.v2+json response.header.content-length=527 response.header.content-type=application/vnd.docker.distribution.manifest.v2+json response.header.date="Wed, 30 Sep 2020 17:36:33 GMT" response.header.docker-content-digest="sha256:2ca5e69e244d2da7368f7088ea3ad0653c3ce7aaccd0b8823d11b0d5de956002" response.header.docker-distribution-api-version=registry/2.0 response.header.etag="\"sha256:2ca5e69e244d2da7368f7088ea3ad0653c3ce7aaccd0b8823d11b0d5de956002\"" response.header.strict-transport-security="max-age=31536000" response.status="200 OK" size=527 url="https://registry-1.docker.io/v2/library/busybox/manifests/sha256:2ca5e69e244d2da7368f7088ea3ad0653c3ce7aaccd0b8823d11b0d5de956002"
panic: assignment to entry in nil map

goroutine 335 [running]:
github.com/containerd/cri/pkg/server.appendInfoHandlerWrapper.func1.1(0x2171620, 0xc0004de840, 0xc0009d62c0, 0x34, 0xc0004b8d70, 0x47, 0x20f, 0x0, 0x0, 0x0, ...)
	/home/mike/go/src/github.com/containerd/cri/pkg/server/image_pull.go:485 +0x171
github.com/containerd/cri/vendor/github.com/containerd/containerd/images.HandlerFunc.Handle(0xc0009bde90, 0x2171620, 0xc0004de840, 0xc0009d62c0, 0x34, 0xc0004b8d70, 0x47, 0x20f, 0x0, 0x0, ...)
	/home/mike/go/src/github.com/containerd/cri/vendor/github.com/containerd/containerd/images/handlers.go:55 +0x82
github.com/containerd/cri/vendor/github.com/containerd/containerd.(*unpacker).handlerWrapper.func1.1(0x2171620, 0xc0004de840, 0xc0009d62c0, 0x34, 0xc0004b8d70, 0x47, 0x20f, 0x0, 0x0, 0x0, ...)
	/home/mike/go/src/github.com/containerd/cri/vendor/github.com/containerd/containerd/unpacker.go:292 +0xf6
github.com/containerd/cri/vendor/github.com/containerd/containerd/images.HandlerFunc.Handle(0xc000a00720, 0x2171620, 0xc0004de840, 0xc0009d62c0, 0x34, 0xc0004b8d70, 0x47, 0x20f, 0x0, 0x0, ...)
	/home/mike/go/src/github.com/containerd/cri/vendor/github.com/containerd/containerd/images/handlers.go:55 +0x82
github.com/containerd/cri/vendor/github.com/containerd/containerd/images.Dispatch.func1(0xc0004c2000, 0xc000474000)
	/home/mike/go/src/github.com/containerd/cri/vendor/github.com/containerd/containerd/images/handlers.go:134 +0xd5
github.com/containerd/cri/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0004dca50, 0xc000187480)
	/home/mike/go/src/github.com/containerd/cri/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x59
created by github.com/containerd/cri/vendor/golang.org/x/sync/errgroup.(*Group).Go
	/home/mike/go/src/github.com/containerd/cri/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66


```

Reply:
Hah thank you! :) fixed

Reply:
@mikebrow It just occurred to me that we need to annotate children, and not the image itself. Since the stargz snapshotter Mount call passes in children labels. I made the update. PTAL.